README.md
# Minimal Onboarding

[![Version](https://img.shields.io/pub/v/minimal_onboarding.svg)](https://pub.dartlang.org/packages/minimal_onboarding)
[![Open Source Love](https://badges.frapsoft.com/os/v1/open-source.svg?v=102)](https://opensource.org/licenses/MIT)
[![License](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/xsahil03x/fancy_on_boarding/blob/master/LICENSE)

A quick minimal flutter onboarding

### GIF
<img src="https://raw.github.com/alihussam/minimal_onboarding/master/minimal_onboarding.gif" height="400" alt="GIF"/>

## Installation
Add following line in the `dependencies:` section of `pubspec.yaml` file:

```
minimal_boarding: <latest_version>
```
### Import following class
```dart
import 'package:minimal_boarding/minimal_boarding.dart';
```

### Create a list of OnboardingPageModel

```dart
 final onboardingPages = [
    OnboardingPageModel( 'assets/hotels1.png', 'Title 1', 'Lorem ipsum ...'),
    OnboardingPageModel( 'assets/hotels2.png', 'Title 2', 'Lorem ipsum ...'),
    OnboardingPageModel( 'assets/hotels3.png', 'Title 3', 'Lorem ipsum ...'),
];
```

### Use the created list in MinimalOnboarding
```dart
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: MinimalBoarding(
        onboardingPages: onboardingPages,
        dotsDecoration: DotsDecorator(
            activeColor: Color(0xFF4E67EB),
            size: const Size.square(9.0),
            activeSize: const Size(18.0, 9.0),
            activeShape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(5.0)),
        ),
        onFinishButtonPressed: () =>
            Navigator.of(context).pushReplacementNamed('/home'),
        onSkipButtonPressed: () =>
            Navigator.of(context).pushReplacementNamed('/home'),
      ),
    );
  }
```

# Dependencies
Minimal Onboarding depends on the use of following package(s):
```
dots_indicator: 1.2.0
```

# License

    Copyright (c) 2018 Sahil Kumar
    
    Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
    
    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
    
    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [1.0.0]

- Add title in pages
- Option to add custom decoration for dot progress indicator
  
## [0.0.2]

- Fix Onboarding App bar color issue
- Update Example
- Update Readme.md and add Gif

## [0.0.1] - 2020-07-13

- Initial Release


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:minimal_onboarding/minimal_onboarding.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  List<OnboardingPageModel> onboardingPages = [
    OnboardingPageModel('images/page1.png', 'Onboarding Page 1',
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at est rhoncus, posuere enim mollis, accumsan odio. '),
    OnboardingPageModel('images/page2.png', 'Onboarding Page 2',
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at est rhoncus, posuere enim mollis, accumsan odio. '),
    OnboardingPageModel('images/page3.png', 'Onboarding Page 3',
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at est rhoncus, posuere enim mollis, accumsan odio. '),
  ];

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Minimal Onboarding example',
      home: MinimalOnboarding(
          onboardingPages: onboardingPages,
          dotsDecoration: DotsDecorator(
            activeColor: Color(0xFF4E67EB),
            size: const Size.square(9.0),
            activeSize: const Size(18.0, 9.0),
            activeShape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(5.0)),
          ),
          onFinishButtonPressed: () => print('Onboarding finished')),
    );
  }
}


lib/src/onboarding_page.dart
import 'package:flutter/material.dart';
import 'package:minimal_onboarding/src/onboarding_page_model.dart';

/// Create a single onboarding pageView slide
class OnboardingPage extends StatelessWidget {
  final OnboardingPageModel _page;
  OnboardingPage(this._page);

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        Expanded(child: Center(child: Image.asset(_page.imagePath))),
        Container(
          height: 150,
          padding: EdgeInsets.symmetric(horizontal: 20),
          child: Column(
            children: <Widget>[
              Text(_page.title,
                  style: TextStyle(fontWeight: FontWeight.bold),
                  textAlign: TextAlign.center),
              SizedBox(height: 20),
              Text(_page.text, textAlign: TextAlign.center),
            ],
          ),
        ),
      ],
    );
  }
}


lib/src/minimal_onboarding.dart
import 'dart:async';
import 'package:dots_indicator/dots_indicator.dart';
import 'package:flutter/material.dart';
import 'package:minimal_onboarding/src/onboarding_page.dart';
import 'package:minimal_onboarding/src/onboarding_page_model.dart';

/// Entry Point Class to create an onboarding page
class MinimalOnboarding extends StatefulWidget {
  final List<OnboardingPageModel> onboardingPages;
  final VoidCallback onFinishButtonPressed;
  final VoidCallback onSkipButtonPressed;
  final String finishButtonText;
  final String skipButtonText;
  final String previousPageButtonText;
  final String nextPageButtonText;
  final bool showSkipButton;
  final Color color;
  final DotsDecorator dotsDecoration;
  static const _defaultColor = Color(0xFF4E67EB);
  static const _dotsDecoration = DotsDecorator(activeColor: _defaultColor);

  MinimalOnboarding({
    @required this.onboardingPages,
    @required this.onFinishButtonPressed,
    this.onSkipButtonPressed,
    this.finishButtonText = 'Done',
    this.skipButtonText = 'Skip',
    this.previousPageButtonText = 'Back',
    this.nextPageButtonText = 'Next',
    this.showSkipButton = true,
    this.color = _defaultColor,
    this.dotsDecoration = _dotsDecoration,
  }) : assert(onboardingPages.length != 0 && onFinishButtonPressed != null);

  @override
  _MinimalOnboardingState createState() => _MinimalOnboardingState();
}

class _MinimalOnboardingState extends State<MinimalOnboarding> {
  PageController _pageController;
  StreamController<double> _streamController;

  @override
  void initState() {
    _streamController = StreamController<double>.broadcast();
    _pageController = PageController();
    _pageController
        .addListener(() => _streamController.add(_pageController.page));
    super.initState();
  }

  @override
  void dispose() {
    super.dispose();
    _streamController.close();
  }

  previousPage() => _pageController.previousPage(
      duration: Duration(milliseconds: 400), curve: Curves.easeIn);
  nextPage() => _pageController.nextPage(
      duration: Duration(milliseconds: 400), curve: Curves.easeIn);

  @override
  Widget build(BuildContext context) {
    // create app bar
    final _appBar = AppBar(
      elevation: 0.0,
      backgroundColor: Colors.transparent,
      title: StreamBuilder(
        stream: _streamController.stream,
        initialData: 0.0,
        builder: (context, snapshot) {
          return DotsIndicator(
            dotsCount: widget.onboardingPages.length,
            position: snapshot.data,
            decorator: widget.dotsDecoration,
          );
        },
      ),
      actions: widget.showSkipButton
          ? [
              FlatButton(
                child: Text(widget.skipButtonText),
                onPressed: widget.onSkipButtonPressed,
              ),
            ]
          : [],
    );

    // Create bottom navigation bar
    final _bottomNavigation = Container(
      height: 60,
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: <Widget>[
          Container(
            child: StreamBuilder(
              stream: _streamController.stream,
              initialData: 0.0,
              builder: (context, snapshot) {
                if (snapshot.data < 1) return Container();
                return FlatButton(
                  child: Text(widget.previousPageButtonText),
                  onPressed: previousPage,
                );
              },
            ),
          ),
          Container(
            child: StreamBuilder(
              stream: _streamController.stream,
              initialData: 0.0,
              builder: (context, snapshot) {
                return RaisedButton(
                    color: widget.color,
                    shape: RoundedRectangleBorder(
                        borderRadius:
                            BorderRadius.only(topLeft: Radius.circular(18.0))),
                    child: Text(
                      snapshot.data > widget.onboardingPages.length - 2
                          ? widget.finishButtonText
                          : widget.nextPageButtonText,
                      style: TextStyle(color: Colors.white),
                    ),
                    onPressed: snapshot.data > widget.onboardingPages.length - 2
                        ? widget.onFinishButtonPressed
                        : nextPage);
              },
            ),
          ),
        ],
      ),
    );

    // create body
    final _body = Container(
      child: PageView(
          controller: _pageController,
          children: List.generate(
              widget.onboardingPages.length,
              (index) =>
                  OnboardingPage(widget.onboardingPages.elementAt(index)))),
    );

    // build and return main body
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: _appBar,
      bottomNavigationBar: _bottomNavigation,
      body: _body,
    );
  }
}
